odl/space/ntuples.py

```diff
@@ -499,19 +501,19 @@ def fallback_copy(x1, x2, n):
         _lincomb(a+b, x1, 0, x1, out, dtype)
     elif out is x1 and out is x2:
         # All the vectors are aligned -> out = (a+b)*out
-        scal(a+b, out.data, out.size)
+        scal(a+b, out.data, native(out.size))
```

Comment:
I don't see a performance hit by casting a single integer. And so far, the BLAS functions seem to be the only ones so far which are "stupid" enough to screw up if the input size is not a native integer. I wonder what happens if one passes a `numpy.int32`.

Other than that, it's once again about making code more predictable across Py2/3. With integers, it's mostly about printing (with or w/o `L` for long ones), but also about instance checks (which we don't have so far). See [here](http://python-future.org/compatible_idioms.html#long-integers).